### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by jukzi**

### CODE REVIEW

**Comment by jukzi** [comment]
```
Error: ] Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for eclipse.jdt.debug:eclipse.jdt.debug:4.31.0-SNAPSHOT: The following artifacts could not be resolved: org.eclipse:eclipse-platform-parent:pom:4.31.0-SNAPSHOT (absent): Could not find artifact org.eclipse:eclipse-platform-parent:pom:4.31.0-SNAPSHOT and 'parent.relativePath' points at wrong local POM @ line 19, column 11
 @ 
Error:  The build could not read 1 project -> [Help 1]
Error:    
Error:    The project eclipse.jdt.debug:eclipse.jdt.debug:4.31.0-SNAPSHOT (/home/runner/work/eclipse.jdt.debug/eclipse.jdt.debug/pom.xml) has 1 error
Error:      Non-resolvable parent POM for eclipse.jdt.debug:eclipse.jdt.debug:4.31.0-SNAPSHOT: The following artifacts could not be resolved: org.eclipse:eclipse-platform-parent:pom:4.31.0-SNAPSHOT (absent): Could not find artifact org.eclipse:eclipse-platform-parent:pom:4.31.0-SNAPSHOT and 'parent.relativePath' points at wrong local POM @ line 19, column 11 -> [Help 2]
```

**Comment by jukzi** [comment]
I could need some help here: why does the maven command [1] works with jdt.core [2] but fails with jdt.ui or jdt.debug?
`Could not find artifact org.eclipse:eclipse-platform-parent:pom:4.31.0-SNAPSHOT`
@akurtakov any idea ?

[1] https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blame/9dd978a9529696cef1ba1eb1700c24ff273d2dab/.github/workflows/codeQLworkflow.yml#L82
`mvn --batch-mode -V -U -ntp -Dcompare-version-with-baselines.skip=true -Pbree-libs -DskipTests=true --fail-at-end clean verify`
[2] https://github.com/eclipse-jdt/eclipse.jdt.core/actions/runs/7887538988/job/21523009449?pr=2013

**Comment by akurtakov** [comment]
Most likely due to the fact that jdt.core passes -Pbuild-individual-bundles to the maven invocation via https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/.mvn/maven.config but jdt.debug doesn't.

File: Jenkinsfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ pipeline {
 				wrap([$class: 'Xvnc', useXauthority: true]) {
 					sh """
 					mvn clean verify --batch-mode --fail-at-end -Dmaven.repo.local=$WORKSPACE/.m2/repository \
-						-Pbuild-individual-bundles -Ptest-on-javase-21 -Pbree-libs -Papi-check -Pjavadoc\
+						-Pbuild -Ptest-on-javase-21 -Pbree-libs -Papi-check -Pjavadoc\
```
Comment (akurtakov) [inline Jenkinsfile:21]: This change is wrong. Whatever is passed via maven.config would better be removed entirely here , aka this -Pbuild should be deleted as there is no such profile at all AFAICT.

Comment (jukzi) [reply Jenkinsfile]: damn typo

**Pull request merged**